src/components/Header.js

```diff
@@ -1,106 +1,197 @@
-import React, { useState } from 'react';
-import Logo from '../assets/images/Logo.png'
-import { Link } from 'react-router-dom';
+import React, { useState } from "react";
+import Logo from "../assets/images/Logo.png";
+import { Link } from "react-router-dom";
 
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faSearch, faBars } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faSearch, faBars } from "@fortawesome/free-solid-svg-icons";
 
-import '../assets/styles/components/Header.css';
+import "../assets/styles/components/Header.css";
 
-import { student } from './mock/student';
+import { student } from "./mock/student";
 
-function Header(){
-    const [state, setState] = useState({
-        isAuth: true,
-        searchInput: '',
-        isSearchCollapsed: true,
-        isMenuCollapsed: true,
-        currentStudent: student[0],
-        isProfileTooltipCollapsed: true
-    });
+function Header() {
+  const [state, setState] = useState({
+    isAuth: true,
+    searchInput: "",
+    isSearchCollapsed: true,
+    isMenuCollapsed: true,
+    currentStudent: student[0],
+    isProfileTooltipCollapsed: true,
+  });
 
-    const handleChange = e => {
-        setState(prevState => ({...prevState, searchInput: e.target.value}))
-        console.log(state.currentStudent.profilePhoto);
-    }
+  const handleChange = (e) => {
+    setState((prevState) => ({ ...prevState, searchInput: e.target.value }));
+    console.log(state.currentStudent.profilePhoto);
+  };
 
-    const toggleSearchCollapse = () => {
-        if(!state.isMenuCollapsed) setState(prevState => ({...prevState, isMenuCollapsed: !state.isMenuCollapsed}));
-        setState(prevState => ({...prevState, isSearchCollapsed: !state.isSearchCollapsed}));
+  const toggleSearchCollapse = () => {
+    if (!state.isMenuCollapsed)
+      setState((prevState) => ({
+        ...prevState,
+        isMenuCollapsed: !state.isMenuCollapsed,
+      }));
+    setState((prevState) => ({
+      ...prevState,
+      isSearchCollapsed: !state.isSearchCollapsed,
+    }));
+  };
 
-    }
+  const toggleMenuCollapse = () => {
+    if (!state.isSearchCollapsed)
+      setState((prevState) => ({
+        ...prevState,
+        isSearchCollapsed: !state.isSearchCollapsed,
+      }));
+    setState((prevState) => ({
+      ...prevState,
+      isMenuCollapsed: !state.isMenuCollapsed,
+    }));
+  };
 
-    const toggleMenuCollapse = () => {
-        if(!state.isSearchCollapsed) setState(prevState => ({...prevState, isSearchCollapsed: !state.isSearchCollapsed}));
-        setState(prevState => ({...prevState, isMenuCollapsed: !state.isMenuCollapsed}));
-    }
+  const SignOut = () => {
+    setState((prevState) => ({
+      ...prevState,
+      isAuth: !state.isAuth,
```

Comment:
use prevState or state not both. I recommend to use prevState because you are using it in other lines.